src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -299,10 +299,10 @@ string_exprt string_constraint_generatort::add_axioms_from_int(
       sum=new_sum;
     }
 
+    equal_exprt premise=res.axiom_for_has_length(size);
     exprt a5=conjunction(digit_constraints);
-    axioms.push_back(a5);
+    axioms.push_back(implies_exprt(premise, a5));
```

Comment:
While you're at it, could you also clarify what that `sum/10` is for axiom `a5` please?